ca/ca.go

```diff
@@ -427,13 +432,61 @@ func (ca *CertificateAuthorityImpl) IssuePrecertificate(ctx context.Context, iss
 		return nil, err
 	}
 
-	precertDER, err := ca.issuePrecertificateInner(ctx, issueReq, serialBigInt, validity, precertType)
-	if err != nil {
-		return nil, err
+	if features.Enabled(features.PrecertificateOCSP) {
+		serialHex := core.SerialToString(serialBigInt)
+		nowNanos := ca.clk.Now().UnixNano()
+		expiresNanos := validity.NotAfter.UnixNano()
+		_, err = ca.sa.AddSerial(ctx, &sapb.AddSerialRequest{
+			Serial:  &serialHex,
+			RegID:   issueReq.RegistrationID,
+			Created: &nowNanos,
+			Expires: &expiresNanos,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		precertDER, err := ca.issuePrecertificateInner(ctx, issueReq, serialBigInt, validity, precertType)
+		if err != nil {
+			return nil, err
+		}
+
+		ocspResp, err := ca.GenerateOCSP(ctx, core.OCSPSigningRequest{
+			CertDER: precertDER,
+			Status:  "good",
```

Comment:
minor nit: you can use the `core.OCSPStatusGood` const from the `core` package to avoid the `"good"` magic value.